'_tcslen' undefined in Windows build

The Windows CI reports an warning:
```
..\src\gmt_support.c(14907): warning C4013: '_tcslen' undefined; assuming extern returning int
```
probably because we forgot to include the header file `tchar.h` somewhere.

--- a/src/gmt_support.c
+++ b/src/gmt_support.c
@@ -87,6 +87,7 @@
 #include <glob.h>
 #else
 #include <Windows.h>
+#include <tchar.h>
 #endif
 
 /*! . */

--- a/src/gmt_support.c
+++ b/src/gmt_support.c
@@ -14905,7 +14905,7 @@ char *gmt_putusername (struct GMT_CTRL *GMT) {
 #ifdef WIN32
 	{
 		char name[GMT_LEN256] = {""}, *U = NULL;
-		DWORD Size = _tcslen (name);
+		DWORD Size = (DWORD)_tcslen (name);
 		if (GetUserName (name, &Size)) /* Got a user name, return it */
 			return (strdup (name));
 		if (U = getenv ("USERNAME"))	/* Got a name from the environment instead */

